source/guides/references/protractor-migration-guide.md

```diff
@@ -0,0 +1,565 @@
+
+### Time travel through tests
+
+Rather than trying to decipher errors inside of your terminal, with Cypress you can debug your tests just like you would debug your application normally. In addition, Cypress allows you to see snapshots of your Angular application as the test is executed so you can debug your applications more easily.
+
+### Easily automate screenshots
+
+Another powerful feature of end-to-end testing is the ability to capture screenshots and recording. While possible in both tools, Protractor requires you to manually tap into Node.js and has limited configuration options.
+
+**Before: Protractor**
+
+```javascript
+const fs = require('fs')
+
+describe('Dashboard', () => {
+  it('should render dashboard', () => {
+    browser.get('/dashboard')
+    browser.takeScreenshot().then((png) => {
+      const stream = fs.createWriteStream(filename)
+
+      writeScreenShot(png, 'dashboard.png')
+      stream.write(new Buffer(png, 'base64'))
+      stream.end()
+    })
+  })
+})
+```
+
+**After: Cypress**
+
+```javascript
+describe('Dashboard', () => {
+  it('should render dashboard', () => {
+	  cy.visit('/dashboard')
+	  cy.screenshot('dashboard.png')
+  })
+})
+```
+
+For more information on what configuration options are available, be sure to check out the [official Cypress documentation on the screenshot command](https://docs.cypress.io/api/commands/screenshot.html).
+
+## FAQs
+
+### Do I have to replace all of my tests with Cypress immediately?
+
+Absolutely not. While it might sound ideal to replace Protractor entirely, we know that most projects and teams still need to deliver features to customers and it's unrealistic to halt the team's progress.
+
+### Can Protractor and Cypress coexist in the same app?
+
+Yes! Your Protractor tests would continue to live in the `e2e` directory that Angular CLI scaffolded while all Cypress tests would live in a sibling folder named `cypress`.
+
+```
+.
+├── cypress
+├── e2e
+├── src
+├── .editorconfig
+├── .gitignore
+├── angular.json
+├── browserslist
+├── cypress.json
+├── karma.conf.js
+├── package.json
+├── README.md
+├── tsconfig.app.json
+├── tsconfig.json
+├── tsconfig.spec.json
+└── tslint.json
+```
+
+In fact, as you work through migrating to Cypress, we believe that progressively enhancing your e2e tests with Cypress is the best path forward to ensure that feature development is not impacted.
+
+## Getting Started
+
+Like any library you would add to your application, we start by adding it as a dependency to our `package.json`.
+
+```
+npm install cypress -D
+```
+
+Then, since Cypress can run in parallel with your application, let's install Concurrently to simplify our npm script.
+
+```
+npm install concurrently -D
+```
```

Comment:
specify shell language for correct styling.

```suggestion
```shell
npm install concurrently -D
```
```